### DISCUSSION TITLE
Auto-imports not working with tailwind example turborepo

### DISCUSSION THREAD

**Discussion by viveleroi** [vercel/turborepo#7251] [Help] [answered]
I created a new app with the [tailwind demo](https://github.com/vercel/turbo/tree/main/examples/with-tailwind) and am toying around in vscode. I can't get auto-imports working - for example when trying to import a component from the `packages/ui` package from inside the `app/web` application, VSCode won't show the component in the auto-import list.

I'm using the settings. I have never worked with monorepos before, and I assume it's a tsconfig error, but what should I look at to resolve this? 

Looking into this further, it seems like an issue with the package.json `exports` field.

This works, VSCode detects and allows me to import `Button`:

```
"exports": {
    "./button": "./src/components/button.tsx"
}
```

But this doesn't:

```
"exports": {
    "./components/*": "./src/components/*.tsx"
}
```

I even tried `"./src/components/**/*.tsx"` with no luck. I don't want to have to define an export for every single component I add...

**Comment by anthonyshew** [comment, marked as answer]
This is intentional behavior from the TypeScript team from what I understand: https://github.com/microsoft/TypeScript/issues/53116#issuecomment-1458887175

TL;DR: Allowing wildcard auto-imports could have bad performance implications.
